code/modules/cargo/export_scanner.dm

```diff
@@ -1,22 +1,159 @@
+				return
+			else
+				payments_acc = potential_acc.registered_account
+				playsound(src, 'sound/machines/ping.ogg', 40, TRUE)
+				to_chat(user, span_notice("[src] registers the ID card. Tag a wrapped item to create a barcode."))
+		else if(!potential_acc.registered_account)
+			to_chat(user, span_warning("This ID card has no account registered!"))
+			return
+	if(istype(attacking_item, /obj/item/paper))
+		if (!(paper_count >= max_paper_count))
+			paper_count += 10
+			qdel(attacking_item)
+			if (paper_count >= max_paper_count)
+				paper_count = max_paper_count
+				to_chat(user, span_notice("[src]'s paper supply is now full."))
+				return
+			to_chat(user, span_notice("You refill [src]'s paper supply, you have [paper_count] left."))
+		else
+			to_chat(user, span_notice("[src]'s paper supply is full."))
 
+/obj/item/export_scanner/attack_self_secondary(mob/user, modifiers)
+	. = ..()
+	if(scanning_mode == SCAN_SALES_TAG)
+		if(paper_count <= 0)
+			to_chat(user, span_warning("You're out of paper!'."))
+			return
+		if(!payments_acc)
+			to_chat(user, span_warning("You need to swipe [src] with an ID card first."))
+			return
+		paper_count -= 1
+		playsound(src, 'sound/machines/click.ogg', 40, TRUE)
+		to_chat(user, span_notice("You print a new barcode."))
+		var/obj/item/barcode/new_barcode = new /obj/item/barcode(src)
+		new_barcode.payments_acc = payments_acc		// The sticker gets the scanner's registered account.
+		new_barcode.cut_multiplier = cut_multiplier		// Also the registered percent cut.
+		user.put_in_hands(new_barcode)
+	if(scanning_mode == SCAN_PRICE_TAG)
+		if(loc != user)
+			to_chat(user, span_warning("You must be holding \the [src] to continue!"))
+			return
+		var/chosen_price = tgui_input_number(user, "Set price", "Price", new_custom_price)
+		if(!chosen_price || QDELETED(user) || QDELETED(src) || !user.canUseTopic(src, be_close = TRUE, no_dexterity = FALSE, no_tk = TRUE) || loc != user)
+			return
+		new_custom_price = chosen_price
+		to_chat(user, span_notice(" The [src] will now give things a [new_custom_price] cr tag."))
+
+/obj/item/export_scanner/CtrlClick(mob/user)
+	. = ..()
+	if(scanning_mode == SCAN_SALES_TAG)
+		payments_acc = null
+		to_chat(user, span_notice("You clear the registered account."))
+
+/obj/item/export_scanner/AltClick(mob/user)
+	. = ..()
+	if(scanning_mode == SCAN_SALES_TAG)
+		var/potential_cut = input("How much would you like to pay out to the registered card?","Percentage Profit ([round(cut_min*100)]% - [round(cut_max*100)]%)") as num|null
+		if(!potential_cut)
+			cut_multiplier = initial(cut_multiplier)
+		cut_multiplier = clamp(round(potential_cut/100, cut_min), cut_min, cut_max)
+		to_chat(user, span_notice("[round(cut_multiplier*100)]% profit will be received if a package with a barcode is sold."))
+
+/obj/item/export_scanner/examine(mob/user)
+	. = ..()
+	. += span_notice("[src] has [paper_count]/[max_paper_count] available barcodes. Refill with paper.")
```

Comment:
i think "It" will be better than the object name, youre already examining the object